diff -ruN a/lib/unixTerminal.js b/lib/unixTerminal.js
--- a/lib/unixTerminal.js
+++ b/lib/unixTerminal.js
@@ -28,7 +28,7 @@
 var pty = native.module;
 var helperPath = native.dir + '/spawn-helper';
 helperPath = path.resolve(__dirname, helperPath);
-helperPath = helperPath.replace('app.asar', 'app.asar.unpacked');
-helperPath = helperPath.replace('node_modules.asar', 'node_modules.asar.unpacked');
+helperPath = helperPath.replace(path.sep + 'app.asar' + path.sep, path.sep + 'app.asar.unpacked' + path.sep);
+helperPath = helperPath.replace(path.sep + 'node_modules.asar' + path.sep, path.sep + 'node_modules.asar.unpacked' + path.sep);
 var DEFAULT_FILE = 'sh';
 var DEFAULT_NAME = 'xterm';
diff -ruN a/src/unixTerminal.ts b/src/unixTerminal.ts
--- a/src/unixTerminal.ts
+++ b/src/unixTerminal.ts
@@ -16,7 +16,7 @@
 let helperPath = native.dir + '/spawn-helper';
 helperPath = path.resolve(__dirname, helperPath);
-helperPath = helperPath.replace('app.asar', 'app.asar.unpacked');
-helperPath = helperPath.replace('node_modules.asar', 'node_modules.asar.unpacked');
+helperPath = helperPath.replace(`${path.sep}app.asar${path.sep}`, `${path.sep}app.asar.unpacked${path.sep}`);
+helperPath = helperPath.replace(`${path.sep}node_modules.asar${path.sep}`, `${path.sep}node_modules.asar.unpacked${path.sep}`);

 const DEFAULT_FILE = 'sh';
 const DEFAULT_NAME = 'xterm';
