diff --git a/out/codeSign/macCodeSign.js b/out/codeSign/macCodeSign.js
index 3bde9755d703a70fc3a45c048c249559fef78d82..9db6a66ee59dbe28a167683b791d835766f279dd 100644
--- a/out/codeSign/macCodeSign.js
+++ b/out/codeSign/macCodeSign.js
@@ -155,16 +155,16 @@ async function createKeychain({ tmpDir, cscLink, cscKeyPassword, cscILink, cscI
     if (cscIKeyPassword != null) {
         cscPasswords.push(cscIKeyPassword);
     }
-    return await importCerts(keychainFile, certPaths, cscPasswords);
+    return await importCerts(keychainFile, certPaths, cscPasswords, keychainPassword);
 }
-async function importCerts(keychainFile, paths, keyPasswords) {
+async function importCerts(keychainFile, paths, keyPasswords, keychainPassword) {
     var _a;
     for (let i = 0; i < paths.length; i++) {
         const password = (_a = keyPasswords[i]) !== null && _a !== void 0 ? _a : "";
         await (0, builder_util_1.exec)("/usr/bin/security", ["import", paths[i], "-k", keychainFile, "-T", "/usr/bin/codesign", "-T", "/usr/bin/productbuild", "-P", password]);
         // https://stackoverflow.com/questions/39868578/security-codesign-in-sierra-keychain-ignores-access-control-settings-and-ui-p
         // https://github.com/electron-userland/electron-packager/issues/701#issuecomment-322315996
-        await (0, builder_util_1.exec)("/usr/bin/security", ["set-key-partition-list", "-S", "apple-tool:,apple:", "-s", "-k", password, keychainFile]);
+        await (0, builder_util_1.exec)("/usr/bin/security", ["set-key-partition-list", "-S", "apple-tool:,apple:", "-s", "-k", keychainPassword, keychainFile]);
     }
     return {
         keychainFile,
diff --git a/templates/nsis/installSection.nsh b/templates/nsis/installSection.nsh
--- a/templates/nsis/installSection.nsh
+++ b/templates/nsis/installSection.nsh
@@ -11,6 +11,13 @@ StrCpy $appExe "$INSTDIR\${APP_EXECUTABLE_FILENAME}"
 # must be called before uninstallOldVersion
 !insertmacro setLinkVars
 
+Var /GLOBAL keepShortcuts
+StrCpy $keepShortcuts "false"
+
+!ifmacrodef customUpdateInstallShouldRun
+  !insertmacro customUpdateInstallShouldRun $R7
+!endif
+
 !ifdef ONE_CLICK
   !ifdef HEADER_ICO
     File /oname=$PLUGINSDIR\installerHeaderico.ico "${HEADER_ICO}"
@@ -38,8 +45,12 @@ StrCpy $appExe "$INSTDIR\${APP_EXECUTABLE_FILENAME}"
   ${endif}
 !endif
-
-Var /GLOBAL keepShortcuts
-StrCpy $keepShortcuts "false"
+
+!ifmacrodef customUpdateInstallShouldRun
+  ${if} $R7 == "true"
+    !insertmacro customUpdateInstall
+    Goto appBuilderInstallSectionDone
+  ${endif}
+!endif
 !insertMacro setIsTryToKeepShortcuts
 ${if} $isTryToKeepShortcuts == "true"
   ReadRegStr $R1 SHELL_CONTEXT "${INSTALL_REGISTRY_KEY}" KeepShortcuts
@@ -108,3 +119,5 @@ StrCpy $keepShortcuts "false"
     !insertmacro doStartApp
   ${endIf}
 !endif
+
+appBuilderInstallSectionDone:
