diff --git a/lib/client.js b/lib/client.js
index 63d3f59..e2c7c9a 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -1849,6 +1849,7 @@ window.__ModuleLoader__.load({
 			};
 			return (0, react_jsx_runtime.jsxs)("div", {
 				className: clsx(WorkspaceBrowser_module_css_default.root, !wide && WorkspaceBrowser_module_css_default.rail),
+				"data-dsh-workspace-drop-target": "",
 				children: [
 					(0, react_jsx_runtime.jsxs)("div", {
 						className: WorkspaceBrowser_module_css_default.sectionHeader,
