diff --git a/lib/types-CNjZgO4h.js b/lib/types-CNjZgO4h.js
index b220efde20bba91b172345f10d261cffdb887a62..528121ed0a5c59762f309abcf5ba0947ac157cd7 100644
--- a/lib/types-CNjZgO4h.js
+++ b/lib/types-CNjZgO4h.js
@@ -857 +857,2 @@ function spawnSandboxed(api, token, options) {
-		dwFlags: 256,
+		dwFlags: 257,
+		wShowWindow: 0,
@@ -994 +995,2 @@ function spawnSandboxedInherited(api, token, options) {
-		dwFlags: 256,
+		dwFlags: 257,
+		wShowWindow: 0,
